virttest/staging/utils_memory.py

```diff
@@ -24,17 +24,23 @@ def read_from_meminfo(key, session=None):
     func = process.getoutput
     if session:
         func = session.cmd_output
-    meminfo = func('grep %s /proc/meminfo' % key)
-    return int(re.search(r'\d+', meminfo).group(0))
+    search_file = '/proc/meminfo'
```

Comment:
search_file = '/sys/devices/system/node/node%s/meminfo' if node_id else '/proc/meminfo'
maybe this one line can handle line 27~29

Reply:
update